src/shared/util.js

```diff
@@ -16,8 +16,8 @@ export function _toString (val: any): string {
  * If the conversion fails, return original string.
  */
 export function toNumber (val: string): number | string {
-  const n = parseFloat(val, 10)
-  return (n || n === 0) ? n : val
+  const n = parseFloat(val)
+  return isNaN(n) ? val : n
```

Comment:
I don't think using a regex is a good idea. parseFloat seems to be the fastest way to convert a string to a number: https://jsperf.com/number-vs-plus-vs-parsefloat